### PR TITLE
Changes to support broker validation changes done in BrokerValidator

### PR DESCRIPTION
**Pull request by shahzaibj**
Related Common PR: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1191

### CODE REVIEW

**Comment by shahzaibj** [comment]
The tests are failing because we are lacking a shadow for HttpClient (it is already being worked on by @gitaumoses4)

File: msal/src/main/java/com/microsoft/identity/client/internal/api/BrokerClientIdRefreshTokenAccessor.java [hunk 1]
```diff
@@ -49,6 +50,25 @@
 
     private static final String TAG = BrokerClientIdRefreshTokenAccessor.class.getSimpleName();
 
+    private static void throwIfNotValidBroker(final Context context) throws MsalClientException {
+        final BrokerValidator brokerValidator = new BrokerValidator(context);
+        final Set<BrokerData> validBrokers = brokerValidator.getValidBrokers();
+
+        for (final BrokerData brokerData : validBrokers) {
+            if (brokerData.packageName.equals(context.getPackageName())) {
```
Comment (AdamBJohnsonx) [inline msal/src/main/java/com/microsoft/identity/client/internal/api/BrokerClientIdRefreshTokenAccessor.java:58]: I'm thinking we want a getBrokerData(packageName) method here.

File: msal/src/main/java/com/microsoft/identity/client/internal/api/BrokerClientIdRefreshTokenAccessor.java [hunk 2]
```diff
@@ -49,6 +50,25 @@
 
     private static final String TAG = BrokerClientIdRefreshTokenAccessor.class.getSimpleName();
 
+    private static void throwIfNotValidBroker(final Context context) throws MsalClientException {
```
Comment (iamgusain) [inline msal/src/main/java/com/microsoft/identity/client/internal/api/BrokerClientIdRefreshTokenAccessor.java:53]: >private static void throwIfNotValidBroker [](start = 4, length = 41)

nit: Should this be moved down along with the other private method?

File: msal/src/main/java/com/microsoft/identity/client/internal/api/BrokerClientIdRefreshTokenAccessor.java [hunk 3]
```diff
@@ -59,18 +79,12 @@
      * @return an RT, if there's any.
      * @throws MsalClientException if the calling app is not a broker app.
      */
-    public static @Nullable String get(@NonNull final Context context,
-                                       @NonNull final String accountObjectId) throws MsalClientException {
+    public static @Nullable
+    String get(@NonNull final Context context,
+               @NonNull final String accountObjectId) throws MsalClientException {
         final String methodName = "getBrokerRefreshToken";
```
Comment (iamgusain) [inline msal/src/main/java/com/microsoft/identity/client/internal/api/BrokerClientIdRefreshTokenAccessor.java:85]: > methodName = "getBrokerRefreshToken" [](start = 20, length = 37)

 isn't the method name "get" here or "BrokerClientIdRefreshTokenAccessor.get". 
Any idea why it is set as  "getBrokerRefreshToken"

File: msal/src/main/java/com/microsoft/identity/client/internal/api/BrokerClientIdRefreshTokenAccessor.java [hunk 4]
```diff
@@ -49,6 +50,25 @@
...
     private static final String TAG = BrokerClientIdRefreshTokenAccessor.class.getSimpleName();
 
+    private static void throwIfNotValidBroker(final Context context) throws MsalClientException {
+        final BrokerValidator brokerValidator = new BrokerValidator(context);
+        final Set<BrokerData> validBrokers = brokerValidator.getValidBrokers();
+
+        for (final BrokerData brokerData : validBrokers) {
+            if (brokerData.packageName.equals(context.getPackageName())) {
+                if (!brokerValidator.verifySignature(context.getPackageName())) {
+                    throw new MsalClientException(NOT_ELIGIBLE_TO_USE_BROKER, "This can only be invoked by Broker apps with a valid signature hash.");
```
Comment (iamgusain) [inline msal/src/main/java/com/microsoft/identity/client/internal/api/BrokerClientIdRefreshTokenAccessor.java:60]: >NOT_ELIGIBLE_TO_USE_BROKER [](start = 50, length = 26)

nit: shouldn't this be something like "INVALID_BROKER_SIGNATURE_HASH" and the below one be "INVALID_BROKER_APP"

File: msal/src/main/java/com/microsoft/identity/client/internal/api/BrokerClientIdRefreshTokenAccessor.java (hunk 1 above)
Comment (shahzaibj) [reply msal/src/main/java/com/microsoft/identity/client/internal/api/BrokerClientIdRefreshTokenAccessor.java]: Addressed here: https://github.com/AzureAD/microsoft-authentication-library-for-android/pull/1274/commits/31f24b52a4d1e8dc035473a18c2c552b9e3b7b9d

File: msal/src/main/java/com/microsoft/identity/client/internal/api/BrokerClientIdRefreshTokenAccessor.java (hunk 2 above)
Comment (shahzaibj) [reply msal/src/main/java/com/microsoft/identity/client/internal/api/BrokerClientIdRefreshTokenAccessor.java]: Addressed

File: msal/src/main/java/com/microsoft/identity/client/internal/api/BrokerClientIdRefreshTokenAccessor.java (hunk 3 above)
Comment (shahzaibj) [reply msal/src/main/java/com/microsoft/identity/client/internal/api/BrokerClientIdRefreshTokenAccessor.java:63]: No idea...probably a typo here

Comment (shahzaibj) [reply msal/src/main/java/com/microsoft/identity/client/internal/api/BrokerClientIdRefreshTokenAccessor.java:63]: or may be a pattern that we have been following that I'm not aware of

File: msal/src/main/java/com/microsoft/identity/client/internal/api/BrokerClientIdRefreshTokenAccessor.java (hunk 4 above)
Comment (shahzaibj) [reply msal/src/main/java/com/microsoft/identity/client/internal/api/BrokerClientIdRefreshTokenAccessor.java]: This comment is no longer valid as I've refactored the code...but the idea is that at this layer we just know if the broker is valid or not...we don't know why...resolving comment for now

File: msal/src/main/java/com/microsoft/identity/client/internal/api/BrokerClientIdRefreshTokenAccessor.java (hunk 3 above)
Comment (shahzaibj) [reply msal/src/main/java/com/microsoft/identity/client/internal/api/BrokerClientIdRefreshTokenAccessor.java:63]: I'll keep this is as is for now....we can fix it later in a separate PR if needed as I believe we've followed this pattern in a number of places

**Review by AdamBJohnsonx — APPROVED** [review]

**Pull request merged**